File: android/app/src/main/java/com/dam/starwarsapp/domain/model/VideoCandidate.kt
package com.dam.starwarsapp.domain.model

/**
 * A generic candidate returned by a video provider when searching for a title.
 */
data class VideoCandidate(
    val provider: String,
    val contentId: String,
    val title: String,
    val watchUrl: String,
    val thumbnailUrl: String? = null,
)

File: iOS/Star-Wars-Demo-APP/Star-Wars-Demo-APP/Domain/Video/VideoCandidate.swift
import Foundation

public struct VideoCandidate: Equatable {
    public let provider: String
    public let contentId: String
    public let title: String?
    public let watchURL: URL?
    public let thumbnailURL: URL?

    public init(provider: String, contentId: String, title: String? = nil, watchURL: URL? = nil, thumbnailURL: URL? = nil) {
        self.provider = provider
        self.contentId = contentId
        self.title = title
        self.watchURL = watchURL
        self.thumbnailURL = thumbnailURL
    }
}
